### PR TITLE
EOS-14275: Add tracepoints for dstore_obj_ops operations (dsal changes)

### PR DESCRIPTION
**Pull request by shahidSTX**
# cortx-dsal Change Summary

## Problem Statement
_[EOS-14275](https://jts.seagate.com/browse/EOS-14275):_
_ADDB: Add tracepoints for dstore_obj_ops operations_

## Problem Description
_Add ADDB tracepoints in DSAL layer for dstore_obj_ops_

## Solution Overview
_Add trace points in all necessary dstore related functions_

## Unit Test Cases
_Full build, executed testing scripts_

## Testing
<details>
  <summary>Click here to see the test execution output</summary>

```
[root@ssc-vm-c-0689 cortx-posix]# ./scripts/test.sh
 --
 --
Configuration Completed
Clean indexes prepared
NSAL Unit tests
NS Tests
Test results are logged to /var/log/cortx/test/ut/ut_nsal.log
Total tests  = 5
Tests passed = 5
Tests failed = 0

Iterator test
Test results are logged to /var/log/cortx/test/ut/ut_nsal.log
Total tests  = 2
Tests passed = 2
Tests failed = 0

KVTree test
Test results are logged to /var/log/cortx/test/ut/ut_nsal.log
Total tests  = 17
Tests passed = 17
Tests failed = 0


CORTXFS Unit tests
Endpoint ops Tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.logs
Total tests  = 4
Tests passed = 4
Tests failed = 0

FS Tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 3
Tests passed = 3
Tests failed = 0

Directory tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 14
Tests passed = 14
Tests failed = 0

File creation tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 4
Tests passed = 4
Tests failed = 0

Link Tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 6
Tests passed = 6
Tests failed = 0

Rename tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 3
Tests passed = 3
Tests failed = 0

Attribute Tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 5
Tests passed = 5
Tests failed = 0

Xattr file Tests
Test results are logged to /var/log/cortx/test/ut/xattr_file_ops.log
Total tests  = 9
Tests passed = 8
Tests failed = 1

Xattr dir Tests
Test results are logged to /var/log/cortx/test/ut/xattr_dir_ops.log
Total tests  = 9
Tests passed = 8
Tests failed = 1

IO tests
Test results are logged to /var/log/cortx/test/ut/ut_cortxfs.log
Total tests  = 7
Tests passed = 7
Tests failed = 0

DSAL Unit tests
Dsal basic test
Test results are logged to /var/log/cortx/test/ut/ut_dsal.log
Total tests  = 11
Tests passed = 11
Tests failed = 0

Dsal IO test
Test results are logged to /var/log/cortx/test/ut/ut_dsal.log
Total tests  = 1
Tests passed = 1
Tests failed = 0

Dsal space stats test
Test results are logged to /var/log/cortx/test/ut/ut_dsal.log
Total tests  = 2
Tests passed = 2
Tests failed = 0
```

</details>

## Checklist
- [x] **Compilation:** _This patch does not break compilation_
- [x] **Merge conflicts:** _None_
- [ ] **Code review:** _In progress_
- [X] **Sanity Testing:** _Done_
- [ ] **Documentation:** _TBD_

### CODE REVIEW

File: src/dstore/dstore_base.c [hunk 1]
```diff
@@ -59,6 +59,8 @@ int dstore_init(struct collection_item *cfg, int flags)
 
 	assert(dstore && cfg);
 
+	perfc_trace_inii(PFT_DSTORE_INIT, PEM_DSTORE_TO_NFS);
+
```
Comment (pratyush-seagate) [inline src/dstore/dstore_base.c:63]: Suggestion: do we need to trace flags?

File: src/dstore/dstore_base.c [hunk 2]
```diff
@@ -134,6 +164,8 @@ static int dstore_obj_shrink(struct dstore_obj *obj,  size_t old_size,
 	off_t offset;
 	char *tmp_buf = NULL;
 
+	perfc_trace_inii(PFT_DSTORE_OBJ_SHRINK, PEM_DSTORE_TO_NFS);
+
```
Comment (pratyush-seagate) [inline src/dstore/dstore_base.c:168]: Suggestion: do we need to trace old_size and new_size?

File: src/dstore/dstore_base.c [hunk 3]
```diff
@@ -184,13 +216,19 @@ static int dstore_obj_shrink(struct dstore_obj *obj,  size_t old_size,
...
 	return rc;
 }
 
 int dstore_obj_resize(struct dstore_obj *obj, size_t old_size, size_t new_size)
 {
 	int rc = 0;
 
+	perfc_trace_inii(PFT_DSTORE_OBJ_RESIZE, PEM_DSTORE_TO_NFS);
+
```
Comment (pratyush-seagate) [inline src/dstore/dstore_base.c:231]: Suggestion: do we need to trace old_size and new_size?

File: src/dstore/dstore_base.c [hunk 4]
```diff
@@ -291,6 +353,8 @@ static int dstore_io_op_init_and_submit(struct dstore_obj *obj,
 	dassert(op_type == DSTORE_IO_OP_WRITE ||
 		op_type == DSTORE_IO_OP_READ);
 
+	perfc_trace_inii(PFT_DSTORE_IO_OP_INIT_AND_SUBMIT, PEM_DSTORE_TO_NFS);
+
```
Comment (pratyush-seagate) [inline src/dstore/dstore_base.c:357]: Suggestion: do we need to trace op_type?

File: src/dstore/dstore_base.c (hunk 1 above)
Comment (shahidSTX) [reply src/dstore/dstore_base.c:63]: No need because as of now all callers of dstore_init and dsal_init set flag to '0'.

**Comment by pratyush-seagate** [comment]
A generic comment for this change. Looks like Vimal already added the addb traces in the low level dstore apis which are directly calling m0 APIs (ref. cortx_dstore_ops).

File: src/dstore/dstore_base.c (hunk 4 above)
Comment (shahidSTX) [reply src/dstore/dstore_base.c:357]: As of now, only read and write ops are allowed inside dstore_io_op_init_and_submit, and we have traces in both of those functions, so the call flow will be clear. Hence not needed.

File: src/dstore/dstore_base.c (hunk 2 above)
Comment (shahidSTX) [reply src/dstore/dstore_base.c:168]: Might be needed, can you suggest a use case where we would need these sizes?

**Comment by shahidSTX** [comment]
> A generic comment for this change. Looks like Vimal already added the addb traces in the low level dstore apis which are directly calling m0 APIs (ref. cortx_dstore_ops).

Yes, but here we will be tracing dstore object related ops, so this will be different from corx_dstore_ops. Let me know if this is not the case.

File: src/dstore/dstore_base.c (hunk 1 above)
Comment (pratyush-seagate) [reply src/dstore/dstore_base.c:63]: ok

File: src/dstore/dstore_base.c (hunk 2 above)
Comment (pratyush-seagate) [reply src/dstore/dstore_base.c:168]: By observing those two params, we can compare the time consumed by such calls and relate the execution time with the size args.

File: src/dstore/dstore_base.c (hunk 4 above)
Comment (pratyush-seagate) [reply src/dstore/dstore_base.c:357]: ok

**Review by pratyush-seagate — APPROVED** [review]
LGTM, +1

**Review by AtitaShirwaikar — APPROVED** [review]
+1

**Review by sachinpunadikar — APPROVED** [review]
+1

**Pull request merged**